src/parsers.jl

```diff
@@ -0,0 +1,76 @@
+"""
+    parse_input_data(input_data::ImageHDU)
+    parse_input_data(input_data::String; hdu_in = nothing)
+    parse_input_data(input_data::FITS; hdu_in = nothing)
+
+Parse input data and returns an Array and WCS object.
+
+# Arguments
+- `input_data`: image to reproject which can be name of a FITS file,
+                an ImageHDU or a FITS file. 
+- `hdu_in`: used to set HDU to use when more than one HDU is present. 
+"""
+function parse_input_data(input_data::ImageHDU)
+    return read(input_data), WCS.from_header(read_header(input_data, String))[1]
+end
+
+function parse_input_data(input_data::String, hdu_in)
```

Comment:
I think you can still make `hdu_in` optional and default to 1, my comment was more about making it a keyword that throws an error if not set.  I see no point in defaulting a function to an error

Reply:
As these are all internal functions, user won't call it directly. And for making `hud_in` optional, I'm keeping that in the reproject signature. See my comment with the reproject signature.

Reply:
Ok, fine